java/vector/src/test/java/org/apache/arrow/vector/TestDictionaryVector.java

```diff
@@ -576,6 +577,125 @@ public void testUnionEquals() {
     }
   }
 
+  @Test
+  public void testEncodeListSubField() {
+    // Create a new value vector
+    try (final ListVector vector = ListVector.empty("vector", allocator);
+        final ListVector dictionaryVector = ListVector.empty("dict", allocator);) {
+
+      UnionListWriter writer = vector.getWriter();
+      writer.allocate();
+
+      //set some values
+      writeListVector(writer, new int[]{10, 20});
+      writeListVector(writer, new int[]{10, 20});
+      writeListVector(writer, new int[]{10, 20});
+      writeListVector(writer, new int[]{30, 40, 50});
+      writeListVector(writer, new int[]{30, 40, 50});
+      writeListVector(writer, new int[]{10, 20});
+      writer.setValueCount(6);
+
+      UnionListWriter dictWriter = dictionaryVector.getWriter();
+      dictWriter.allocate();
+      writeListVector(dictWriter, new int[]{10, 20, 30, 40, 50});
+      dictionaryVector.setValueCount(1);
+
+      Dictionary dictionary = new Dictionary(dictionaryVector, new DictionaryEncoding(1L, false, null));
+
+      try (final ListVector encoded =
+          (ListVector) DictionaryEncoder.encodeSubField(vector, dictionary)) {
+        // verify indices
+        assertEquals(ListVector.class, encoded.getClass());
+
+        assertEquals(6, encoded.getValueCount());
+        assertEquals("[0,1]", encoded.getObject(0).toString());
```

Comment:
this seems fragile can you compare lists instead?

Reply:
fixed.